assets/js/modules/search-console/components/dashboard/SearchFunnelWidgetGA4/Overview/index.js

```diff
@@ -164,6 +176,29 @@ const Overview = ( {
 		! error &&
 		! showRecoverableAnalytics;
 
+	const { setValue } = useDispatch( CORE_UI );
+	const { triggerOnDemandTour } = useDispatch( CORE_USER );
+	useEffect( () => {
+		if (
+			toursAreOnCooldown ||
+			!! shownTour ||
+			! showGA4 ||
+			dashboardType !== DASHBOARD_TYPE_MAIN
```

Comment:
How about we calculate this boolean outside of the effect? That way we don't need to pass all these dependencies just for this condition, and run it more than necessary.

Reply:
Yes, agree, but if I move it outside of this effect, the complexity of the component exceeds the allowed limit and eslint demands to refactor this component. Maybe let's leave this as is for now?

Reply:
Maybe we can move it to a useSelect instead? If not, then it's okay to keep.

Reply:
Combined `toursAreOnCooldown` and `shownTour` into one hook. Let me know if it looks okay to you.

Reply:
Yep, it's still a single value in the end so that's no problem 👍 Thanks!